File: Euclid/Util/StringUtil.cs
using System.Text;

namespace Euclid.Util
{
    public class StringUtil
    {
        #region Public methods

        public static Encoding GetEncoding()
        {
            return Encoding.GetEncoding("ISO-8859-1");
        }

        public static bool HasWhitelistedCharacters(string str, string allowedChars)
        {
            if (str == null)
            {
                return false;
            }

            for (int i = 0; i < str.Length; i++)
            {
                if (allowedChars.Contains(str[i].ToString()))
                {
                    continue;
                }

                return false;
            }

            return true;
        }

        #endregion
    }
}

File: euclid/src/main/java/euclid/util/StringUtil.java
package euclid.util;

import java.nio.charset.Charset;
import java.nio.charset.StandardCharsets;

public class StringUtil {

    public static Charset getEncoding() {
        return StandardCharsets.ISO_8859_1;
    }

    public static boolean hasWhitelistedCharacters(String str, String allowedChars) {
        if (str == null) {
            return false;
        }

        for (int i = 0; i < str.length(); i++) {
            if (allowedChars.indexOf(str.charAt(i)) >= 0) {
                continue;
            }

            return false;
        }

        return true;
    }
}
